src/Controller/Consent.php

```diff
@@ -22,25 +22,30 @@ class Consent
 {
     public const DEMO_CONSENT_URI = 'https://account-d.docusign.com/oauth/auth';
     public const CONSENT_URI = 'https://account.docusign.com/oauth/auth';
+    public const GRANT_TYPE_AUTHORIZATION_CODE = 'authorization_code';
+    public const GRANT_TYPE_IMPLICIT = 'implicit';
 
     private $envelopeBuilder;
     private $router;
     private $consentUri;
     private $integrationKey;
+    private $grantType;
 
-    public function __construct(EnvelopeBuilderInterface $envelopeBuilder, RouterInterface $router, string $consentUri, string $integrationKey)
+    public function __construct(EnvelopeBuilderInterface $envelopeBuilder, RouterInterface $router, string $consentUri, string $integrationKey, string $grantType)
     {
         $this->envelopeBuilder = $envelopeBuilder;
         $this->router = $router;
         $this->consentUri = $consentUri;
         $this->integrationKey = $integrationKey;
+        $this->grantType = $grantType;
     }
 
     public function __invoke(): RedirectResponse
     {
         return new RedirectResponse(sprintf(
-            '%s?response_type=token&scope=signature%%20impersonation&client_id=%s&redirect_uri=%s',
+            '%s?response_type=%s&scope=signature%%20impersonation&client_id=%s&redirect_uri=%s',
             $this->consentUri,
+            self::GRANT_TYPE_AUTHORIZATION_CODE === $this->grantType ? 'code' : 'token',
```

Comment:
Might be better using an array of associated types, it'll be easier to extends later and it avoid a ternary operation. 

```suggestion
           $this->responseType,
```